--- boost/utility/in_place_factory.hpp	Пт июл 16 01:19:14 2010
+++ boost/utility/in_place_factory.hpp	Вт фев  5 03:47:23 2013
@@ -31,6 +31,11 @@
 #else
 #define N BOOST_PP_ITERATION()
 
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4512)
+#endif // #if defined(_MSC_VER)
+
 #if N
 template< BOOST_PP_ENUM_PARAMS(N, class A) >
 #endif
@@ -67,6 +72,10 @@
   BOOST_PP_REPEAT(N, BOOST_DEFINE_INPLACE_FACTORY_CLASS_MEMBER_DECL, _)
 };
 
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif // #if defined(_MSC_VER)
+
 #if N > 0
 template< BOOST_PP_ENUM_PARAMS(N, class A) >
 inline BOOST_PP_CAT(in_place_factory,N)< BOOST_PP_ENUM_PARAMS(N, A) >

--- boost/utility/base_from_member.hpp	Вс ноя 28 06:27:25 2004
+++ boost/utility/base_from_member.hpp	Вт фев  5 03:25:55 2013
@@ -62,6 +62,11 @@
 
 // Contributed by Daryle Walker
 
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4512)
+#endif // #if defined(_MSC_VER)
+
 template < typename MemberType, int UniqueID = 0 >
 class base_from_member
 {
@@ -77,6 +82,10 @@
 
 };  // boost::base_from_member
 
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif // #if defined(_MSC_VER)
+
 }  // namespace boost
 
 
@@ -84,4 +93,4 @@
 #undef BOOST_PRIVATE_CTR_DEF
 
 
-#endif  // BOOST_UTILITY_BASE_FROM_MEMBER_HPP
+#endif  // BOOST_UTILITY_BASE_FROM_MEMBER_HPP

--- boost/thread/future.hpp	Вс дек 16 23:01:45 2012
+++ boost/thread/future.hpp	Вт фев  5 03:27:53 2013
@@ -351,7 +351,18 @@
 #if defined BOOST_THREAD_PROVIDES_INTERRUPTIONS
                   if(rethrow && thread_was_interrupted)
                   {
+
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4913)
+#endif
+
                       throw boost::thread_interrupted();
+
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
                   }
 #endif
                   if(rethrow && exception)

--- boost/program_options/options_description.hpp	Вс апр  8 14:02:26 2012
+++ boost/program_options/options_description.hpp	Вт фев  5 03:25:51 2013
@@ -170,6 +170,12 @@
         See @ref a_adding_options "here" for option adding interface discussion.
         @sa option_description
     */
+
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4512)
+#endif // #if defined(_MSC_VER)
+
     class BOOST_PROGRAM_OPTIONS_DECL options_description {
     public:
         static const unsigned m_default_line_length;
@@ -260,6 +266,10 @@
 
     };
 
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif // #if defined(_MSC_VER)
+
     /** Class thrown when duplicate option description is found. */
     class BOOST_PROGRAM_OPTIONS_DECL duplicate_option_error : public error {
     public:
@@ -271,4 +281,4 @@
 #   pragma warning (pop)
 #endif
 
-#endif
+#endif

